baseclasses/BaseSolver.py

```diff
@@ -115,14 +56,14 @@ def setOption(self, name, value):
         Parameters
         ----------
         name : str
-           Name of option to set. Not case sensitive
+           Name of option to set. Not case sensitive.
         value : varies
            Value to set. Type is checked for consistency.
 
         """
-        name = name.lower()
+        # Check if the option exists
         try:
-            self.defaultOptions[name]
+            default = self.defaultOptions[name]
```

Comment:
Can unpack this here too, so that you don't have to refer to `default[0]` etc later on.